cmdlets/sqlserver/Invoke-MiLinkCutover.ps1

```diff
@@ -0,0 +1,221 @@
+function Invoke-MiLinkCutover {
+    <#
+    .SYNOPSIS
+    Cmdlet that performs cutover from SQL Server (primary) to SQL managed Instance (secondary)
+    
+    .DESCRIPTION
+    Cmdlet that performs cutover from SQL Server (primary) to SQL managed Instance (secondary)
+    Cmdlet can be ran in interactive mode or you can provide all necessary parameters
+    Cmdlet supports ShouldProcess, can be dry-ran with -WhatIf parameter
+    Cmdlet consists of 4 steps:
+        1. switch replication mode to sync
+        2. compare lsns
+        3. remove the link
+        4. remove ags
+
+    .PARAMETER ResourceGroupName
+    Resource group name of Managed Instance 
+    
+    .PARAMETER ManagedInstanceName
+    Managed Instance Name
+    
+    .PARAMETER SqlInstance
+    Sql Server name
+    
+    .PARAMETER DatabaseName
+    Database name
+    
+    .PARAMETER PrimaryAvailabilityGroup
+    Primary availability group name
+    
+    .PARAMETER SecondaryAvailabilityGroup
+    Secondary availability group name
+    
+    .PARAMETER LinkName
+    Instance link name
+    
+    .PARAMETER ManagedInstanceCredential
+    Managed Instance Credential
+    
+    .PARAMETER CleanupPreference
+    One of { "KEEP_BOTH", "DELETE_DAG", "DELETE_AG_AND_DAG" }
+    Defines actions on SQL Server upon deleting the link
+    
+    .EXAMPLE
+    # Remove-Module Invoke-MiLinkCutover
+    # Import-Module 'C:\{pathtoscript}\Invoke-MiLinkCutover.ps1'
+
+    Invoke-MiLinkCutover -ResourceGroupName CustomerExperienceTeam_RG -ManagedInstanceName chimera-ps-cli-v2 `
+    -SqlInstance chimera -DatabaseName zz -PrimaryAvailabilityGroup AG_test2 -SecondaryAvailabilityGroup MI_test2 -LinkName DAG_test2  -Verbose 
+    
+    $cred = Get-Credential
+    Invoke-MiLinkCutover -ResourceGroupName CustomerExperienceTeam_RG -ManagedInstanceName chimera-ps-cli-v2 `
+    -SqlInstance chimera -DatabaseName zz -PrimaryAvailabilityGroup AG_test2 -SecondaryAvailabilityGroup MI_test2 `
+    -LinkName DAG_test2 -ManagedInstanceCredential $cred -CleanupPreference "DELETE_AG_AND_DAG" -Verbose
```

Comment:
idea was to have two modes of cmdlet operation, Interactive and NonInteractive.
with interactive, we'll have various prompts to decide which step we want to perform in what way (skip, stop, continue etc)
For noninteractive, since we have multiple steps, a `-Force` flag wouldn't be sufficient as we'd normally want more granular control of the steps so we're introducing params such as CleanupPreference (we'll add forced/planned failover option)